java/maven.hints/src/org/netbeans/modules/maven/hints/pom/UseReleaseOptionHint.java

```diff
@@ -0,0 +1,241 @@
+
+    private static final String TARGET_TAG = "target";
+    private static final String SOURCE_TAG = "source";
+    private static final String RELEASE_TAG = "release";
+
+    private static final Configuration config = new Configuration(UseReleaseOptionHint.class.getName(),
+                TIT_UseReleaseVersionHint(), DESC_UseReleaseVersionHint(), true, Configuration.HintSeverity.WARNING);
+
+    @Override
+    public List<ErrorDescription> getErrorsForDocument(POMModel model, Project prj) {
+
+        List<ErrorDescription> hints = new ArrayList<>();
+
+        Build build = model.getProject().getBuild();
+        if (build != null) {
+            for (Plugin plugin : build.getPlugins()) {
+                if (plugin.getArtifactId().equals("maven-compiler-plugin")) {
+                    if (!isPluginCompatible(plugin)) {
+                        return Collections.emptyList();
+                    }
+                    if (plugin.getConfiguration() != null) {
+                        hints.addAll(createHintsForParent("", plugin.getConfiguration()));
+                    }
+                    if (plugin.getExecutions() != null) {
+                        for (PluginExecution execution : plugin.getExecutions()) {
+                            if (execution.getConfiguration() != null) {
+                                hints.addAll(createHintsForParent("", execution.getConfiguration()));
+                            }
+                        }
+                    }
+                    break;
+                }
+            }
+        }
+
+        Properties properties = model.getProject().getProperties();
+        if (properties != null) {
+            hints.addAll(createHintsForParent("maven.compiler.", properties));
+        }
+
+        return hints;
+    }
+
+    private List<ErrorDescription> createHintsForParent(String prefix, POMComponent parent) {
+
+        int source;
+        int target;
+
+        // property name or an int value
+        String release = null;
+
+        try {
+            String sourceText = parent.getChildElementText(POMQName.createQName(prefix+SOURCE_TAG, true));
+            if (isPropertyName(sourceText)) {
+                release = sourceText;
+                sourceText = getProperty(sourceText, parent.getModel());
+            }
+
+            String targetText = parent.getChildElementText(POMQName.createQName(prefix+TARGET_TAG, true));
+            if (isPropertyName(targetText)) {
+                release = targetText;
+                targetText = getProperty(targetText, parent.getModel());
+            }
+
+            source = Integer.parseInt(sourceText);
+            target = Integer.parseInt(targetText);
+            if (release == null) {
+                release = String.valueOf(target);
+            }
+        } catch (NumberFormatException ignored) {
+            return Collections.emptyList();
+        }
+
+        if (source == target && source >= 9) {
```

Comment:
I think you could go down to 6 - at least with OpenJDK 11 javac reports:

```
  --release <release>
        Compile for a specific VM version. Supported targets: 6, 7, 8, 9, 10, 11
```

Reply:
I actually forgot that it goes lower then 9. But it makes sense since now I remember that I read JDK 20 is going to drop support for 7 and below.

 - JDK 12 dropped support for 6
 - JDK 20 dropped support for 7 -> https://inside.java/2022/09/26/quality-heads-up/

The current implementation will only offer the hint if the values are 9 or greater and match. This implies that the JDK is also 9 or later, which means it is always safe to convert to release.

That is why there is no JDK check. I thought to keep it simple otherwise we would have to update this hint periodically each time a JDK drops a version. You think its worth to cover the EOL versions too with this hint?

Reply:
I also wouldn't know how to query the configured JDK. The method has a `Project` parameter which has a lookup. I could get the `NbMavenProject` which has a bit more context but still no JDK.

Reply:
i checked all registered services in the lookup of `org.netbeans.api.project.Project` but couldn't find anything which would give me access to the configured JDK of the maven project. Going to shelve the JDK check but would be willing to implement it in a followup PR.